app/models/concerns/galtersufia/generic_file/header_lookup.rb

```diff
@@ -11,43 +11,39 @@ module HeaderLookup
       END_SPARQL_MESH_URI = "%27%2C%20%27i%27))%20%0D%0A%7D%20%0D%0AORDER%20BY%20%3Fd%20%0D%0A"
       LCSH_BASE_URI = "http://id.loc.gov/authorities/subjects/suggest/?q="
 
-      def pid_lookup_by_scheme(term, scheme)
-        scheme == :mesh ? mesh_term_pid_lookup(term) : lcsh_term_pid_lookup(term)
+      def pid_lookup_by_scheme(term="", scheme="")
+        if term.blank? || scheme.blank?
+          return
+        elsif scheme == :mesh
+          memoized_mesh_lookups[term] || mesh_term_pid_lookup(term)
+        elsif scheme == :lcsh
+          memoized_lcsh_lookups[term] || lcsh_term_pid_lookup(term)
+        else
+          return
+        end
       end
 
       # return PID for provided mesh_header using SPARQL query
       def mesh_term_pid_lookup(mesh_term="")
-        return unless mesh_term.present?
-
-        result = memoized_mesh_lookups[mesh_term]
-        return result if result.present?
-
-        query_result  = HTTParty.get(BASE_SPARQL_MESH_URI + mesh_term + END_SPARQL_MESH_URI)
-        json_parsed_result = JSON.parse(query_result)
+        hits = perform_and_parse_mesh_query(mesh_term)
 
-        if hits = hits_from_mesh_sparql(json_parsed_result)
+        if hits.present?
           result = pid_from_mesh_hits(hits)
           memoized_mesh_lookups[mesh_term] = result
-          result
         else
-          ["Error Searching"]
+          nil
         end
       end
 
       # return PID for provided lcsh_term
       def lcsh_term_pid_lookup(lcsh_term="")
-        return unless lcsh_term.present?
-
-        result = memoized_lcsh_lookups[lcsh_term]
-        return result if result.present?
-
         subject_names, subject_id_uris = perform_and_parse_lcsh_query(lcsh_term)
 
-        if subject_names && subject_id_uris
+        if subject_names.present? && subject_id_uris.present?
```

Comment:
I think this is the root of the original error. The query was actually returning something, but what we wanted from it was blank. This resolves the actual errors that ruby was throwing so we can do something else when there is no result.